bin/oref0-setup.sh

```diff
@@ -314,17 +314,21 @@ function install_or_upgrade_nodejs () {
         echo "Your installed nodejs ($(node --version)) is very slow to start (took ${NODE_EXECUTION_TIME}s)"
         echo "This is a known problem with certain versions of Raspberry Pi OS."
 
-        if prompt_yn "Install a new nodejs version using nvm?" Y; then
-            echo "Installing nvm and using it to replace the system-provided nodejs"
-    
-            # Download nvm
-            curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.39.0/install.sh | bash
-            # Run nvm, adding its aliases to this shell
-            source ~/.nvm/nvm.sh
-            # Use nvm to install nodejs
-            nvm install 10.24.1
-            # Symlink node into /usr/local/bin, where it will shadow /usr/bin/node
-            ln -s ~/.nvm/versions/node/v10.24.1/bin/node /usr/local/bin/node
+        if prompt_yn "Install a new nodejs version using n?" Y; then
+            if [[ ! $(command -v n) ]]; then
+                echo "n already exists on the system, using it to install a new version of node..."
+                n 10
+            else
+                echo "Installing n and using it to replace the system-provided nodejs"
+                echo "Installing node via n..."
+                curl -L https://raw.githubusercontent.com/tj/n/master/bin/n -o n
+                # Install the latest version of node 10.x.x
+                sudo bash n 10
+                # Delete the local n binary used to boostrap the install
+                rm n
+                # Install n globally
+                sudo npm install -g n
+            fi
```

Comment:
I just realized that I don't know how this code will work when node (e.g. from package manager) is already installed. Despite saying it will, it doesn't replace/remove the existing binary. Would `sudo apt remove nodejs npm` or some `dpkg` command work? I would feel uncomfortable leaving a package manager version and trying to write over it, though it's possible that the installed version isn't from the package manager (e.g., what if it's already from nvm?). Or do we need to do something a bit more hacky like what was done before with the symlink on old line 327? This will likely need to change before it can merge.

Reply:
It's installing to /usr/local/, so it probably won't overwrite apt's install which iirc installs to /usr/bin/

Reply:
Latest changes include an attempt to uninstall apt and nvm versions. I don't know if that is sufficient, but I would like to think so. Testing would be good.